extensions/ql-vscode/src/view/jest.setup.ts

```diff
@@ -15,6 +15,11 @@ Object.defineProperty(window, "matchMedia", {
   })),
 });
 
+// Used by Primer React
+(window as any).CSS = {
+  supports: jest.fn().mockResolvedValue(false),
```

Comment:
What are we mocking here exactly? If this always returns false, are we saying we won't support Primer CSS? My primer-fu is failing me on this one. 

I can't see primer/css installed in package.json so I'm assuming it's a dependency of `primer/react` or some other `@primer/*` package. 

I'm also aware we want to move away from Primer entirely and it looks like only one component uses it, through `ThemeProvider`. I'm wondering if we could make a copy of the `ThemeProvider` component and delete primer entirely. One for later probably.

Reply:
We're mocking `CSS.supports` here to make these calls work: https://github.com/primer/react/blob/f0e0ec77f5d91c73ce22d10bc8d55a1a5ab2a9e6/src/PageLayout/useStickyPaneHeight.ts#L106-L113. We're essentially saying that we don't support any modern CSS features, see [the MDN documentation for `CSS.supports`](https://developer.mozilla.org/en-US/docs/Web/API/CSS/supports). This won't change any of the behaviour, it just makes it look like we're using an older browser (which `jest-dom` essentially is since it doesn't support CSS at all).

The `ThemeProvider` code doesn't look like it uses many other files from `@primer/react`: https://github.com/primer/react/blob/f0e0ec77f5d91c73ce22d10bc8d55a1a5ab2a9e6/src/ThemeProvider.tsx. However, we'd also need to copy over the `Overlay` component. This seems to have a lot of dependencies on other files in `@primer/react`, so we'd also need to copy over those: https://github.com/primer/react/blob/f0e0ec77f5d91c73ce22d10bc8d55a1a5ab2a9e6/src/Overlay.tsx

@charisk is working on a new CodePaths view that doesn't use the `Overlay` component, so we can get rid of Primer once that work is done. That would also allow us to get rid of this mock.